spec/sec-class-definitions-patch.html

```diff
@@ -53,225 +47,334 @@ <h1>Static Semantics: Early Errors</h1>
+
+    <emu-grammar>ArrowFunction : ArrowParameters `=>` ConciseBody</emu-grammar>
     <emu-alg>
-      1. <ins>Return the TopLevelVarDeclaredNames of |StatementList|.</ins>
+      1. Return *false*.
     </emu-alg>
-  </emu-clause>
 
-  <emu-clause id="sec-class-definitions-static-semantics-varscopeddeclarations">
-    <h1>Static Semantics: VarScopedDeclarations</h1>
-    <emu-grammar collapsed><ins>ClassBlockStatementList: [empty]</ins></emu-grammar>
+    <emu-grammar>
+      AsyncArrowFunction : `async` AsyncArrowBindingIdentifier `=>` AsyncConciseBody
+    </emu-grammar>
     <emu-alg>
-      1. <ins>Return a new empty List.</ins>
+      1. Return *false*.
     </emu-alg>
-    <emu-grammar collapsed><ins>ClassBlockStatementList: StatementList</ins></emu-grammar>
+
+    <emu-grammar>
+      AsyncArrowFunction : CoverCallExpressionAndAsyncArrowHead `=>` AsyncConciseBody
+    </emu-grammar>
+    <emu-alg>
+      1. Return *false*.
+    </emu-alg>
+
+    <emu-grammar>
+      ForInOfStatement : `for` `await` `(` LeftHandSideExpression `of` AssignmentExpression `)` Statement
+
+      ForInOfStatement : `for` `await` `(` `var` ForBinding `of` AssignmentExpression `)` Statement
+
+      ForInOfStatement : `for` `await` `(` ForDeclaration `of` AssignmentExpression `)` Statement
+    </emu-grammar>
     <emu-alg>
-      1. <ins>Return the TopLevelVarScopedDeclarations of |StatementList|.</ins>
+      1. Return *true*.
     </emu-alg>
   </emu-clause>
 
   <emu-clause id="sec-runtime-semantics-classstaticblockdefinitionevaluation" aoid="ClassStaticBlockDefinitionEvaluation">
     <h1><ins>Runtime Semantics: ClassStaticBlockDefinitionEvaluation</ins></h1>
     <p>With parameter _homeObject_.</p>
-    <emu-grammar collapsed>ClassStaticBlock: `static` `{` ClassStaticBlockBody `}`</emu-grammar>
+    <emu-grammar>ClassStaticBlock: `static` `{` ClassStaticBlockBody `}`</emu-grammar>
     <emu-alg>
       1. Let _lex_ be the running execution context's LexicalEnvironment.
       1. Let _privateScope_ be the running execution context's PrivateEnvironment.
-      1. Let _body_ be OrdinaryFunctionCreate(~Method~, &laquo; &raquo;, |ClassStaticBlockBody|, _lex_, _privateScope_).
+      1. Let _sourceText_ be the empty sequence of Unicode code points.
+      1. Let _body_ be OrdinaryFunctionCreate(%Function.prototype%, _sourceText_, &laquo; &raquo;, |ClassStaticBlockBody|, ~non-lexical-this~, _lex_, _privateScope_).
       1. Perform MakeMethod(_body_, _homeObject_).
       1. Return the ClassStaticBlockDefinition Record { [[Body]]: _body_ }.
     </emu-alg>
+    <emu-note>The function created for _body_ is never directly accessible to ECMAScript code.</emu-note>
   </emu-clause>
 
-  <emu-clause id="sec-runtime-semantics-classelementevaluation" aoid="ClassElementEvaluation">
+  <emu-clause id="sec-static-semantics-classelementevaluation" type="sdo" aoid="ClassElementEvaluation">
```

Comment:
It's odd that this has an anchor that says "static semantics" but is titled "Runtime Semantics". However, I checked the existing Ecma-262 text and this is also the case there, so I guess the best practice would be to conform to that.

Reply:
I'm not sure if you're asking me to change something here. The change in this PR came from copying the clause from the ECMA262 spec.

Reply:
@littledan: Was intended for `ClassElementEvaluation` to be identified as `sec-static-semantics-`, even though it's labeled with `<h1>Runtime Semantics: ...</h1>`?

Reply:
I've updated the header to use `sec-runtime-semantics-` since that seems more appropriate. We can address it further in the ECMA-262 PR